ncurses/0002-fix-double-free-on-unknown-terminal.patch

```diff
@@ -0,0 +1,13 @@
+--- a/ncurses/tinfo/lib_setup.c.orig	2017-07-01 20:24:50.000000000 +0200
++++ a/ncurses/tinfo/lib_setup.c	2018-02-17 17:53:49.304266500 +0100
+@@ -722,8 +722,8 @@
+ 	    termp->Filedes = (short) Filedes;
+ 	    termp->_termname = strdup(tname);
+ 	} else {
+-	    _nc_free_termtype2(&TerminalType(termp));
+-	    free(my_tcb);
++	    //_nc_free_termtype2(&TerminalType(termp));
++	    //free(my_tcb);
```

Comment:
Maybe we can set `termp = NULL;` after it was released (same for `my_tcb`) and then avoid calling `TerminalType(termp)` altogether if `termp` is `NULL` (or probably just guard the first line behind `if (termp)`)?

Reply:
IIUC, this won't work.  The first `free` happens a few calls deep and only the pointer value is passed, so the pointers in the calling function here can't be affected by it.

Reply:
It's worsened by all the aliasing (the above-allocated TERMINAL_CONTROL_BLOCK is pointed to by `termp`, `my_tcb` *and* `TCB` inside this function). :/

Reply:
It appears that there's a fix for this issue upstream: https://github.com/ThomasDickey/ncurses-snapshots/commit/dae8e7bc .

Reply:
It's eerily similar to my work-around. :D